recipes/master.rb

```diff
@@ -20,6 +20,15 @@
   action :enable
 end
 
+user node['salt']['master']['api']['user']['name'] do
```

Comment:
I am inclined to make the user setup optional to avoid such an issue rather than left it to a wrapper cookbook. If cookbook requires a user to function, user resource should be included in the cookbook itself with option to enable/disable.